### PR TITLE
fix(conf) fix sql in escalation conf

### PR DESCRIPTION
**Pull request by a-launois**
## Description

Fixes MON-13452 and MON-13449

fix sql requests and data display for escalation configuration

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [x] 21.04.x
- [x] 21.10.x
- [x] 22.04.x
- [x] 22.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/include/configuration/configObject/escalation/DB-Func.php [hunk 1]
```diff
@@ -40,540 +41,617 @@
...
-        #Duplicate entry
-    } elseif ($dbResult->rowCount() >= 1 && $esc["esc_id"] != $id) {
-        return false;
-    } else {
-        return true;
-    }
+
+    $id = isset($form) ? $form->getSubmitValue('esc_id') : null;
+
+    $stmt = $pearDB->prepare("SELECT esc_name, esc_id FROM escalation WHERE esc_name = :name");
```
Comment (kduret) [inline www/include/configuration/configObject/escalation/DB-Func.php]:
```suggestion
    $stmt = $pearDB->prepare("SELECT esc_id FROM escalation WHERE esc_name = :name");
```

File: www/include/configuration/configObject/escalation/DB-Func.php [hunk 2]
```diff
@@ -40,540 +41,617 @@
...
+            continue;
+        }
+
+        $params[':contactGroupId' . $key] = $contactGroupId;
+        $queryParams[] = "(:escalationId, :contactGroupId{$key})";
     }
-    $fields["esc_hosts"] = "";
-    if (isset($ret["esc_hosts"])) {
-        $fields["esc_hosts"] = implode(",", $ret["esc_hosts"]);
+
```
Comment (kduret) [inline www/include/configuration/configObject/escalation/DB-Func.php:465]:
```suggestion
    if (empty($params)) {
        return;
    }

```

File: www/include/configuration/configObject/escalation/DB-Func.php [hunk 3]
```diff
@@ -40,540 +41,617 @@
...
+    foreach ($params as $paramName => $value) {
+        $stmt->bindValue($paramName, $value, \PDO::PARAM_INT);
     }
-    $fields["esc_hgs"] = "";
-    if (isset($ret["esc_hgs"])) {
-        $fields["esc_hgs"] = implode(",", $ret["esc_hgs"]);
+    if (count($params)) {
+        $stmt->bindValue(':escalationId', $escalationId, \PDO::PARAM_INT);
+        $stmt->execute();
     }
```
Comment (kduret) [inline www/include/configuration/configObject/escalation/DB-Func.php]:
```suggestion
    $stmt->bindValue(':escalationId', $escalationId, \PDO::PARAM_INT);
    $stmt->execute();
```

File: www/include/configuration/configObject/escalation/DB-Func.php [hunk 4]
```diff
@@ -40,540 +41,617 @@
...
+    $query .= implode(', ', $queryParams);
+
+    $stmt = $pearDB->prepare($query);
+    foreach ($params as $paramName => $value) {
+        $stmt->bindValue($paramName, $value, \PDO::PARAM_INT);
     }
-    $fields["esc_metas"] = "";
-    if (isset($ret["esc_metas"])) {
-        $fields["esc_metas"] = implode(",", $ret["esc_metas"]);
+    if (count($params)) {
```
Comment (kduret) [inline www/include/configuration/configObject/escalation/DB-Func.php]: same as above

File: www/include/configuration/configObject/escalation/DB-Func.php [hunk 5]
```diff
@@ -40,540 +41,617 @@
...
-    $pearDB->query($rq);
-    $ret = array();
-    $ret = CentreonUtils::mergeWithInitialValues($form, 'esc_hosts');
-    for ($i = 0; $i < count($ret); $i++) {
-        $rq = "INSERT INTO escalation_host_relation ";
-        $rq .= "(escalation_esc_id, host_host_id) ";
-        $rq .= "VALUES ";
-        $rq .= "('" . $esc_id . "', '" . $ret[$i] . "')";
-        $pearDB->query($rq);
+    if (count($params)) {
```
Comment (kduret) [inline www/include/configuration/configObject/escalation/DB-Func.php]: same as above

File: www/include/configuration/configObject/escalation/DB-Func.php [hunk 6]
```diff
@@ -40,540 +41,617 @@
...
-        $pearDB->query($rq);
+
+    $query = "INSERT INTO escalation_servicegroup_relation (escalation_esc_id, servicegroup_sg_id) VALUES ";
+    $query .= implode(', ', $queryParams);
+
+    $stmt = $pearDB->prepare($query);
+    foreach ($params as $paramName => $value) {
+        $stmt->bindValue($paramName, $value, \PDO::PARAM_INT);
+    }
+    if (count($params)) {
```
Comment (kduret) [inline www/include/configuration/configObject/escalation/DB-Func.php]: same as above

File: www/include/configuration/configObject/escalation/DB-Func.php [hunk 7]
```diff
@@ -40,540 +41,617 @@
...
-        $pearDB->query($rq);
+
+    $query = "INSERT INTO escalation_service_relation (escalation_esc_id, service_service_id, host_host_id) VALUES ";
+    $query .= implode(', ', $queryParams);
+
+    $stmt = $pearDB->prepare($query);
+    foreach ($params as $paramName => $value) {
+        $stmt->bindValue($paramName, $value, \PDO::PARAM_INT);
+    }
+    if (count($params)) {
```
Comment (kduret) [inline www/include/configuration/configObject/escalation/DB-Func.php]: same as above

File: www/include/configuration/configObject/escalation/DB-Func.php [hunk 8]
```diff
@@ -40,540 +41,617 @@
...
-    $pearDB->query($rq);
-    $ret = array();
-    $ret = CentreonUtils::mergeWithInitialValues($form, 'esc_metas');
-    for ($i = 0; $i < count($ret); $i++) {
-        $rq = "INSERT INTO escalation_meta_service_relation ";
-        $rq .= "(escalation_esc_id, meta_service_meta_id) ";
-        $rq .= "VALUES ";
-        $rq .= "('" . $esc_id . "', '" . $ret[$i] . "')";
-        $pearDB->query($rq);
+    if (count($params)) {
```
Comment (kduret) [inline www/include/configuration/configObject/escalation/DB-Func.php]: same as above

**Review by kduret — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Pull request merged**